main.py

```diff
@@ -55,34 +56,37 @@ def list_models() -> JSONResponse:
     return JSONResponse(content=content)
 
 
-async def map_tools(req_tools: list | None = None) -> list[Tool] | None:
-    function_declarations: list[FunctionDeclaration] = []
-    if req_tools:
-        for tool in req_tools:
-            parameters = tool['function']['parameters']
-            if parameters is None:
-                parameters = {
-                    "properties":
-                        {
-                            "fake":
-                                {
-                                    "description": "a fake description",
-                                    "type": "string"
-                                }
-                        },
-                    "type": "object"
-                }
-
-            function_declarations.append(
-                FunctionDeclaration(
-                    name=tool["function"]["name"],
-                    parameters=parameters,
-                    description=tool["function"]["description"]
-                )
+async def map_tools(req_tools: List | None = None) -> List[Tool] | None:
+    if req_tools is None or len(req_tools) < 1:
+        return None
+
+    function_declarations = []
+    for tool in req_tools:
+        parameters = tool['function']['parameters']
+        if parameters is None:
+            parameters = {
+                "properties":
+                    {
+                        "fake":
+                            {
+                                "description": "a fake description",
+                                "type": "string"
+                            }
+                    },
+                "type": "object"
+            }
+
+        function_declarations.append(
+            FunctionDeclaration(
+                name=tool["function"]["name"],
+                description=tool["function"]["description"],
+                parameters=parameters,
             )
-        tools: list[Tool] = [Tool.from_function_declarations(function_declarations)]
-        return tools
-    return None
+        )
+
+    tools: list["Tool"] = [Tool.from_function_declarations(function_declarations)]
+
+    return tools
```

Comment:
This change is not strictly required. Just figured it made the code a bit more clear. LMK if you want me to drop it.